packages/sanes-chrome-extension/src/extension/background/actions/createPersona/requestCallback.ts

```diff
@@ -59,13 +51,44 @@ export async function getIdentitiesCallback(
   matchingIdentities: ReadonlyArray<PublicIdentity>,
   meta: any, // eslint-disable-line
 ): Promise<ReadonlyArray<PublicIdentity>> {
-  return requestCallback(reason, 'getIdentities', meta, matchingIdentities, []);
+  if (!isRequestMeta(meta)) {
+    throw new Error('Unexpected type of data in meta field');
+  }
+  const { senderUrl } = meta;
+
+  const persona = getCreatedPersona();
+  const chainNames = persona.getChains();
```

Comment:
Even better. Just use `chainNames = (await getConfigurationFile()).names` here and neither Persona nor AccountManager need to know about the chain names.

Reply:
Already did! But we have a downside, introducing an await here, we force the JS thread to continue 
```
 signingServer
    .handleUnchecked(request, meta)
    .then(sendResponse)
    .catch(sendResponse);

  return true;
```

Which is normal, of course, but we would need to re-adapt all tests because we call:
```
handleExternalMessage(request, sender, jest.fn());
expect(RequestHandler.requests().length).toBe(1);
```
After calling `handleExternalMessage` we check the `RequestHandler`, introducing an await as you suggest, the test will execute immediately the second line and therefore it will break.

I can refactor all tests, of course, but in the end, I went with my solution because I saw interesting having encapsulated the chains inside persona. I do not feel really comfortable accessing here to the configuration files, it makes more sense persona be aware of the chains where it operates.

On the other hand, I see your points, why complicate persona and personaManager having the option to solve the situation here? It is a good question. Another benefit I am seeing while I write is maybe we can use the chains in another place as well?

If at the end you decide this change should be done, create a ticket. Let's try to merge this asap.

Reply:
Refactoring this separately makes sense. I see your point.

The reason why we don't need chain names in Persona is that all the logic is based on Chain ID, which are (next week) properly specified and unique. Chain names, are just a user interface addition to make it more convenient for the user to identify which chain is which. The logic would work perfectly fine if you just display chain IDs instead of the names.